internal/iris/data_iris_cases.go

```diff
@@ -0,0 +1,207 @@
+	"github.com/blackstork-io/fabric/plugin/dataspec/constraint"
+	"github.com/blackstork-io/fabric/plugin/plugindata"
+)
+
+func makeIrisCasesDataSource(loader ClientLoadFn) *plugin.DataSource {
+	return &plugin.DataSource{
+		Doc:      "Retrieve cases from Iris API",
+		DataFunc: fetchIrisCasesData(loader),
+		Config: &dataspec.RootSpec{
+			Attrs: []*dataspec.AttrSpec{
+				{
+					Name:        "api_url",
+					Type:        cty.String,
+					Constraints: constraint.RequiredMeaningful,
+					Doc:         "Iris API url",
+				},
+				{
+					Name:        "api_key",
+					Type:        cty.String,
+					Secret:      true,
+					Constraints: constraint.RequiredMeaningful,
+					Doc:         "Iris API Key",
+				},
+				{
+					Name:       "insecure",
+					Type:       cty.Bool,
+					DefaultVal: cty.BoolVal(false),
+					Doc:        "Enable/disable insecure TLS",
+				},
+			},
+		},
+		Args: &dataspec.RootSpec{
+			Attrs: []*dataspec.AttrSpec{
+				{
+					Name: "case_ids",
+					Type: cty.List(cty.Number),
+					Doc:  "List of Case IDs",
+				},
+				{
+					Name: "customer_id",
+					Type: cty.Number,
+					Doc:  "Case Customer ID",
+				},
+				{
+					Name: "owner_id",
+					Type: cty.Number,
+					Doc:  "Case Owner ID",
+				},
+				{
+					Name: "severity_id",
+					Type: cty.Number,
+					Doc:  "Case Severity ID",
+				},
+				{
+					Name: "state_id",
+					Type: cty.Number,
+					Doc:  "Case State ID",
+				},
+				{
+					Name: "soc_id",
+					Type: cty.String,
+					Doc:  "Case SOC ID",
+				},
+				{
+					Name: "start_open_date",
+					Type: cty.String,
+					Doc:  "Case opening date - lower boundary",
+				},
+				{
+					Name: "end_open_date",
+					Type: cty.String,
+					Doc:  "Case opening date - higher boundary",
+				},
+				{
+					Name:       "sort",
+					Type:       cty.String,
+					Doc:        "Sort order",
+					DefaultVal: cty.StringVal("desc"),
+					OneOf: []cty.Value{
+						cty.StringVal("desc"),
+						cty.StringVal("asc"),
+					},
+				},
+				{
+					Name:         "size",
+					Type:         cty.Number,
+					Doc:          "Size limit to retrieve",
+					MinInclusive: cty.NumberIntVal(0),
+					Constraints:  constraint.NonNull,
+					DefaultVal:   cty.NumberIntVal(0),
+				},
+			},
+		},
+	}
+}
+
+func fetchIrisCasesData(loader ClientLoadFn) plugin.RetrieveDataFunc {
+	return func(ctx context.Context, params *plugin.RetrieveDataParams) (plugindata.Data, diagnostics.Diag) {
+		cli, err := parseConfig(params.Config, loader)
+		if err != nil {
+			return nil, diagnostics.Diag{{
+				Severity: hcl.DiagError,
+				Summary:  "Failed to parse configuration",
+			}}
+		}
+		req, err := parseListCasesReq(params.Args)
+		if err != nil {
+			return nil, diagnostics.Diag{{
+				Severity: hcl.DiagError,
+				Summary:  "Failed to parse arguments",
+			}}
+		}
+		size := 0
+		if attr := params.Args.GetAttrVal("size"); !attr.IsNull() {
+			num, _ := attr.AsBigFloat().Int64()
+			size = int(num)
+		}
```

Comment:
Since size is NonNull
```suggestion
		num, _ := params.Args.GetAttrVal("size").AsBigFloat().Int64()
		size := int(num)
```